apps/nextjs/utils/string.ts

```diff
@@ -0,0 +1,6 @@
+const CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+
+export const generateRandomString = (length = 32, chars = CHARS): string =>
```

Comment:
As far as I can tell, those aren't supported in browser. Web crypto API and the (modern) browsers seem to support generating UUIDs, perhaps we can use that?

Reply:
Yeah sounds reasonable, although we'll probably need to polyfill it for the nodejs/nextjs side in case this ends up being called on the server for some reason (or does it have it by default)? In any case, it's probably smart to default to crypto-rng for any secrets 👍

Reply:
On further investigation, it seems the web crypto API has its own drawbacks: browser support isn't guaranteed, and it can only be used in [secure contexts](https://w3c.github.io/webappsec-secure-contexts/). Node supports crypto since v15, but then we'd have to do checks on the context were in and probably use dynamic imports. So I guess a third party library would be the preferred solution. Any suggestions?

Reply:
Web Crypto seems to be 97.19% covered according to https://caniuse.com/cryptography (tracked browsers), which I'd be fine making the cut to, the browsers that don't support it are way old at this point and will likely struggle with other features too (i.e. just [fetch](https://caniuse.com/fetch) has a worse coverage), so I wouldn't consider browser compatibility an issue.

Not sure how secure context affects us here, personally I've yet to hit an issue using web crypto. After looking into it slightly it seems like it offers guarantees we'd like to enforce for security purposes, and overall be a good thing?

Based on my 10 minute investigation now, I'd say web crypto is the way to go here. Doing the swap between Node vs. browser implementation is probably slightly more annoying but doable.

Reply:
Implemented using Web Crypto, falling back to Node's Crypto on server-side.